crel.js

```diff
@@ -40,64 +39,57 @@
             return object instanceof Element;
         },
         appendChild = function(element, child) {
-            if (Array.isArray(child)) {
-                child.map(function(subChild){
-                    appendChild(element, subChild);
-                });
-                return;
-            }
-            if(!crel[isNodeString](child)){
-                child = d.createTextNode(child);
+            if (child) {  // Note: this includes 0
+                if (Array.isArray(child)) {
+                    child.map(function(subChild){
+                        appendChild(element, subChild);
+                    });
+                    return;
+                }
+                if(!crel[isNodeString](child)){
+                    child = d.createTextNode(child);
+                }
+                element.appendChild(child);
             }
-            element.appendChild(child);
+
         };
 
 
-    function crel(element){
-        var args = arguments, //Note: assigned to a variable to assist compilers. Saves about 40 bytes in closure compiler. Has negligable effect on performance.
-            child,
-            settings = args[1],
-            childIndex = 2,
-            argumentsLength = args.length,
-            attributeMap = crel[attrMapString];
+    function crel(element, settings){
+        var args = arguments, //Note: assigned to a variable to assist compilers.
+            index = 1,
+            attributeMap = crel[attrMapString],
+            // Declaring all used variables here shortens the code slightly
+            key,
+            attribute,
+            value;
 
         element = crel[isElementString](element) ? element : d.createElement(element);
-        // shortcut
-        if(argumentsLength > 1){
-            if(!isType(settings,obj) || crel[isNodeString](settings) || Array.isArray(settings)) {
-                --childIndex;
-                settings = null;
-            }
-
-            // shortcut if there is only one child that is a string
-            if((argumentsLength - childIndex) === 1 && isType(args[childIndex], 'string')){
-                element.textContent = args[childIndex];
-            }else{
-                for(; childIndex < argumentsLength; ++childIndex){
-                    child = args[childIndex];
 
-                    if(child !== null){
-                        appendChild(element, child);
+        if (!(!isType(settings, obj) || crel[isNodeString](settings) || Array.isArray(settings))) {
+          index++;
+            for(key in settings){
+                attribute = settings[key];
+                key = attributeMap[key] || key;
+                if (isType(key, fn)) {
+                    key(element, attribute);
+                } else if (isType(attribute, fn) || element[key]) {
+                    if (isType(attribute, obj)) {
+                        // We only check and allow for one level of object depth
+                        for (value in attribute) {

```

Comment:
Isn't crel there to allow creating and editing ALL of the basic attributes and properties of an element? That's why I added the properties manipulation anyways. And with that in place it becomes kind of impossible to edit the style of an element through crel, because you can't access the attribute anymore.
On a side note, it's not specifically a helper for the style property, but then again I don't know if any other property is nested like that.